Pyjion/absint.cpp

```diff
@@ -2541,23 +2541,29 @@ void AbstractInterpreter::unary_negative(int opcodeIndex) {
     auto stackInfo = get_stack_info(opcodeIndex);
     auto one = stackInfo[stackInfo.size() - 1];
 
-    switch (one.Value->kind()) {
-        case AVK_Float:
-            dec_stack();
-            m_comp->emit_unary_negative_float();
-            inc_stack(1, STACK_KIND_VALUE);
-            break;
-        case AVK_Integer:
-            dec_stack();
-            m_comp->emit_unary_negative_tagged_int();
-            inc_stack();
-            break;
-        default:
-            dec_stack();
-            m_comp->emit_unary_negative();
-            error_check("unary negative failed");
-            inc_stack();
-            break;
+    bool handled = false;
+    if (!one.needs_boxing()) {
+        switch (one.Value->kind()) {
+            case AVK_Float:
+                handled = true;
+                dec_stack();
+                m_comp->emit_unary_negative_float();
+                inc_stack(1, STACK_KIND_VALUE);
```

Comment:
The int case we technically have something that we treat as a boxed value on the stack (because we don't know if the int has overflowed at some point).  The float case is always a raw float that's unboxed so we have a value type on the stack